scripts/travis-pr-title-lint

```diff
@@ -0,0 +1,181 @@
+/* eslint max-len: 0 */
+
+const https = require('https');
+
+const shelljs = require('shelljs');
+
+const {
+  TRAVIS_REPO_SLUG,
+  TRAVIS_PULL_REQUEST,
+  VERBOSE,
+} = process.env;
+
+
+const DONT_PANIC_MESSAGE = `
+Don't panic! If your travis build is failing here, please take a look at
+
+ - https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#writing-commit-messages
+
+and feel free to ask for help from one of the maintainers in a comment; we are here to help ;-)
+`;
+
+function findMergeBase() {
+  const res = shelljs.exec('git merge-base HEAD master', {silent: true});
+
+  if (res.code !== 0) {
+    throw new Error(`findMergeBase Error: ${res.stderr}`);
+  }
+
+  const baseCommit = res.stdout.trim();
+  if (VERBOSE === 'true') {
+    console.log('DEBUG findMergeBase:', baseCommit);
+  }
+
+  return baseCommit;
+}
+
+function getGitBranchCommits() {
+  const baseCommit = findMergeBase();
+  const gitCommand = `git rev-list --no-merges HEAD ^${baseCommit}`;
+
+  const res = shelljs.exec(gitCommand, {silent: true});
+
+  if (res.code !== 0) {
+    throw new Error(`getGitBranchCommits Error: ${res.stderr}`);
+  }
+
+  const commits = res.stdout.trim().split('\n');
+  if (VERBOSE === 'true') {
+    console.log('DEBUG getGitBranchCommits:', commits);
+  }
+
+  return commits;
+}
+
+function getGitCommitMessage(commitSha1) {
+  const res = shelljs.exec(`git show -s --format=%B ${commitSha1}`, {silent: true});
+
+  if (res.code !== 0) {
+    throw new Error(`getGitCommitMessage Error: ${res.stderr}`);
+  }
+
+  const commitMessage = res.stdout.trim();
+  if (VERBOSE === 'true') {
+    console.log(`DEBUG getGitCommitMessage: "${commitMessage}"`);
+  }
+
+  return commitMessage;
+}
+
+function getPullRequestTitle() {
+  return new Promise(function(resolve, reject) {
+    const pullRequestURLPath = `/${TRAVIS_REPO_SLUG}/pull/${TRAVIS_PULL_REQUEST}`;
+
+    console.log(`Retrieving the pull request title from https://github.com${pullRequestURLPath}\n`);
+
+    var req = https.get({
+      host: 'github.com',
+      path: pullRequestURLPath,
+      headers: {
+        'User-Agent': 'GitHub... your API can be very annoying ;-)',
+      },
+    }, function(response) {
+      if (response.statusCode < 200 || response.statusCode > 299) {
+        reject(new Error(`getPullRequestTitle got an unexpected statusCode: ${response.statusCode}`));
+        return;
+      }
+
+      var body = '';
+      response.on('data', function(data) {
+        try {
+          body += data;
+
+          // Once we get the closing title tag, we can read
+          // the pull request title and
```

Comment:
The sentence is cut off.

And please use the Github API instead of parsing HTML:

Example: https://api.github.com/repos/mozilla/web-ext/pulls/1518
Documentation: https://developer.github.com/v3/pulls/

Reply:
> The sentence is cut off.

I'll fix the inline comment. 

> And please use the Github API instead of parsing HTML:

No, unfortunately I can't do this, I already know how it is going to end up:

the first version of this script was already using the github API, but then the travis jobs started to fail because of the API usage limits and I had to workaround it this way (See #731, as I mentioned in that PR I wasn't happy about the workaround neither, but it has definitely fixed the intermittency and it worked without any further changes for more than 2 years now, #731 has been merged on Jan 4 2017).